junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/predicates/IsTestableMethod.java

```diff
@@ -46,6 +49,10 @@ public boolean test(Method candidate) {
 		if (isAbstract(candidate)) {
 			return false;
 		}
+		if (returnsVoid(candidate) != mustReturnVoid) {
```

Comment:
BTW, this requested change is not only about readability but more importantly about performance... since the suggested construct benefits from boolean short-circuiting.